fix: prevent submitting an empty form when pressing enter

michielgerritsen opened the pull request:
(I didn't create an issue, sorry for jumpstarting)

Hi,

This fixes something that is annoying me for a long time: When entering or updating a contact field, ie the telephone number, and you press enter, the form is submitted and your data is gone. This fix prevents the form to submit.

### Checklist

#### Before submitting the PR
- [X] Read the [CONTRIBUTING document](https://github.com/monicahq/monica/blob/master/CONTRIBUTING.md) before submitting your PR.
- [X] If the PR is related to an issue or fix one, don't forget to indicate it.
- [X] Indicate `[wip]` in the title of the PR it is is not final yet. Remove `[wip]` when ready. Otherwise the PR will be considered complete and rejected if it's not working.

### General checks
- [X] Make sure that the change you propose is the smallest possible.
- [X] The name of the PR should follow the [conventional commits guideline](https://github.com/monicahq/monica/blob/master/docs/contribute/index.md#conventional-commits) that the project follows.

### Front-end changes
- [ ] If you change the UI, make sure to ask repositories administrators first about your changes by pinging @djaiss or @asbiin in this PR.
- [ ] Screenshots are included if the PR changes the UI.
- [ ] Front-end tests have been written with Cypress.

#### Backend/models changes
- [ ] The API has been updated.
- [ ] API's documentation has been added by submitting a pull request in the [marketing website repository](https://github.com/monicahq/marketing_site/pulls).
- [ ] Tests have been added for the new code.
- [ ] If you change a model, make sure the FakeContentTableSeeder is updated. We need seeders to develop locally and generate fake data.

#### If the code changes the SQL schema
- [ ] Make sure exporting account data as SQL is still working.
- [ ] Make sure your changes do not break importing data with `vCard` and `.csv` files.
- [ ] Make sure account reset and deletion still work.

#### Other tasks
- [X] [CHANGELOG](https://github.com/monicahq/monica/blob/master/CHANGELOG) entry added, if necessary, under `UNRELEASED`.
- [X] [CONTRIBUTORS](https://github.com/monicahq/monica/blob/master/CONTRIBUTORS) entry added, if necessary.
- [X] If it's relevant and worth mentioning, create a changelog entry for this change. The changelog entry will appear inside the UI for all users to see. To know if your change is worth the creation of a changelog entry, [read the documentation](https://github.com/monicahq/monica/blob/master/docs/administrators/tips.md#when-is-it-relevant-to-create-a-changelog-entry).
- [X] Don't forget to [ask for a free account](mailto:regis@monicahq.com) on https://monicahq.com as anyone who contributes can request a free account.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/monicahq/monica?pullRequest=1978) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/monicahq/monica?pullRequest=1978) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/monicahq/monica?pullRequest=1978) it.</sub>

asbiin commented on resources/assets/js/components/people/ContactInformation.vue:
```diff
@@ -35,7 +35,7 @@
 
         <div class="w-100" v-show="contactInformation.edit">
-          <form class="measure center">
+          <form class="measure center" v-on:submit.prevent>
```
We could allow user to hit `Enter`, and just update this form
```suggestion
          <form class="measure center" v-on:submit.prevent="update(contactInformation)">
```

asbiin commented on resources/assets/js/components/people/ContactInformation.vue:
```diff
@@ -56,7 +56,7 @@
 
     <div v-if="addMode">
-      <form class="measure center">
+      <form class="measure center" v-on:submit.prevent>
```
We could allow user to hit `Enter`, and just update this form
```suggestion
          <form class="measure center" v-on:submit.prevent="store">
```

michielgerritsen commented:
Nice addition @asbiin. I have added it. I did also run `yarn run production`, am i supposed to commit the generated `public/*` files? I've added them just in case.

asbiin commented:
Gret @michielgerritsen ! Thank you.

> am i supposed to commit the generated public/* files?

yes, these assets are included in the sources.
